src/libraries/ArrayLib.sol

```diff
@@ -4,24 +4,26 @@ pragma solidity 0.8.21;
 /// @title  ArrayLib
 library ArrayLib {
     function countNonZeroValues(uint16[8] memory arr) internal pure returns (uint8 count) {
-        for (uint256 i; i < arr.length; ++i) {
+        for (uint256 i; i < arr.length; i++) {
             if (arr[i] > 0) ++count;
         }
     }
 
-    function decreaseFirstNValues(uint16[8] storage arr, uint8 numValues, uint16 decrease) internal {
-        for (uint256 i; i < arr.length; ++i) {
+    function decreaseFirstNValues(uint16[8] storage arr, uint8 numValues) internal {
+        for (uint256 i; i < arr.length; i++) {
             if (numValues == 0) return;
 
             if (arr[i] > 0) {
-                arr[i] -= decrease;
+                arr[i] -= 1;
                 numValues--;
             }
         }
+
+        require(numValues == 0, "ArrayLib/invalid-values");
```

Comment:
This makes the method stricter, catching issues with trying to decrease by too much

Reply:
```
 function dfnv(uint8[10] storage arr, uint8 numValues) internal {
        require(numValues <= arr.length, "Out-of-bound");
        if (numValues == 0) return;

        for (uint256 i; i < numValues; i++) {   
            if (arr[i] > 0) {
                arr[i] -= 1;
            }
        }
    }
    ```